Compilation fails on Ubuntu 16.04 ROS kinetic

**Describe the bug**
Following error is seen when I try to compile

[ 32%] Building C object CMakeFiles/apriltag.dir/common/image_u8.c.o
[ 35%] Building C object CMakeFiles/apriltag.dir/common/image_u8x3.c.o
[ 37%] Building C object CMakeFiles/apriltag.dir/common/image_u8x4.c.o
[ 40%] Building C object CMakeFiles/apriltag.dir/common/workerpool.c.o
In file included from /home/locobot1/apriltag_ws/src/apriltag/common/workerpool.c:32:0:
/usr/include/stdio.h:314:6: error: unknown type name ‘_IO_cookie_io_functions_t’
      _IO_cookie_io_functions_t __io_funcs) __THROW __wur;
      ^
CMakeFiles/apriltag.dir/build.make:398: recipe for target 'CMakeFiles/apriltag.dir/common/workerpool.c.o' failed
make[2]: *** [CMakeFiles/apriltag.dir/common/workerpool.c.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/apriltag.dir/all' failed
make[1]: *** [CMakeFiles/apriltag.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
<== Failed to process package 'apriltag': 
  Command '['make', '-j1']' returned non-zero exit status 2

**Operating Sytem**
Ubuntu 16.04

** Installation Method**
Following the locobot auto installation script provided here : https://www.trossenrobotics.com/docs/interbotix_xslocobots/getting_started/user_guide.html

** Code version **
8c45cb387bcc5e4d08b04f01bbc9ba83aae93444

I had an older version of the repo which works fine. The HEAD for the older version:
76482356ff7ecb3f4895b3b600d533b2e65980d9

--- a/.github/workflows/industrial_ci_action.yml
+++ b/.github/workflows/industrial_ci_action.yml
@@ -12,6 +12,7 @@ jobs:
       matrix:
         env:
           # ROS 1
+          - {ROS_DISTRO: kinetic}
           - {ROS_DISTRO: melodic}
           - {ROS_DISTRO: noetic}
           # ROS 2

--- a/.github/workflows/ros_workspace.yml
+++ b/.github/workflows/ros_workspace.yml
@@ -11,10 +11,14 @@ jobs:
 
     strategy:
       matrix:
-        ros_distribution: [noetic, humble]
+        ros_distribution: [melodic, noetic, humble]
         config: ["default"] # nice name
         cmake_args: ['[ ]'] # empty list of options
-        include:
+        include:         
+          - docker_image: ubuntu:18.04
+            ros_distribution: melodic
+            ros_version: 1
+
           - docker_image: ubuntu:20.04
             ros_distribution: noetic
             ros_version: 1
@@ -40,8 +44,8 @@ jobs:
           apt install --no-install-recommends -y git ca-certificates
 
       - uses: actions/checkout@v3
-        with:
-          submodules: recursive
+        #with:
+        #  submodules: recursive
 
       - name: Setup ROS environment
         uses: ros-tooling/setup-ros@v0.3

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,8 @@ add_executable(apriltag_demo example/apriltag_demo.c)
 target_link_libraries(apriltag_demo ${PROJECT_NAME})
 
 # opencv_demo
-find_package(OpenCV COMPONENTS core imgproc videoio highgui QUIET)
+# NB: contrib required for TickMeter in OpenCV 2.4. This is only required for 16.04 backwards compatibility and can be removed in the future.
+find_package(OpenCV COMPONENTS core imgproc videoio highgui contrib QUIET)
 if(OpenCV_FOUND)
     add_executable(opencv_demo example/opencv_demo.cc)
     target_link_libraries(opencv_demo apriltag ${OpenCV_LIBRARIES})

--- a/common/workerpool.c
+++ b/common/workerpool.c
@@ -26,6 +26,7 @@ either expressed or implied, of the Regents of The University of Michigan.
 */
 #include <errno.h>
 
+#define _GNU_SOURCE  // Possible fix for 16.04
 #define __USE_GNU
 #include "common/pthreads_cross.h"
 #include <assert.h>

--- a/example/opencv_demo.cc
+++ b/example/opencv_demo.cc
@@ -26,6 +26,7 @@ either expressed or implied, of the Regents of The University of Michigan.
 */
 
 #include <iostream>
+#include <iomanip>
 
 #include "opencv2/opencv.hpp"
 
@@ -121,9 +122,15 @@ int main(int argc, char *argv[])
     meter.stop();
     cout << "Detector " << famname << " initialized in " 
         << std::fixed << std::setprecision(3) << meter.getTimeSec() << " seconds" << endl;
+#if CV_MAJOR_VERSION > 3
     cout << "  " << cap.get(CAP_PROP_FRAME_WIDTH ) << "x" <<
                     cap.get(CAP_PROP_FRAME_HEIGHT ) << " @" <<
                     cap.get(CAP_PROP_FPS) << "FPS" << endl;
+#else
+    cout << "  " << cap.get(CV_CAP_PROP_FRAME_WIDTH ) << "x" <<
+                    cap.get(CV_CAP_PROP_FRAME_HEIGHT ) << " @" <<
+                    cap.get(CV_CAP_PROP_FPS) << "FPS" << endl;
+#endif
     meter.reset();
 
     Mat frame, gray;

